### DISCUSSION TITLE
Inquire about impact when collecting performance data at short intervals

### DISCUSSION THREAD

**Discussion by MIOKHAN** [NetApp/harvest#1779] [Q&A]
Hi, 

I have a question.
When collecting performance data (cpu, latency, iops) from about 100 NetApp AFF/FAS systems at intervals of 1-3 seconds, is there any performance impact on NetApp storage systems? Are there any references for performance monitoring at such short intervals?

Best Regards.

**Comment by cgrinds** [comment]
hi @MIOKHAN I'll ask around and see what I can find out. A few customers have mentioned to us they collect every 30s, but I don't know of anyone collecting every 1-3 seconds.

Few questions:
- Do your current collections finish within your poll interval?
- You probably already know this, but if you change the Harvest scrape interval, you'll also need to change your Prometheus scrape interval too

**Reply by MIOKHAN** [reply]
This is an inquiry from my customer. The customer knows how to change the interval. However, he wants to know the impact before changing to every 1-3 seconds. Because it is a very performance sensitive system.

**Comment by cgrinds** [comment]
I spoke with some ONTAP engineers about this and they shared the following points:

- There is always some impact to collecting stats. In most cases, that impact is small, but as the frequency increases so does the impact. 
- The amount of impact also depends on how much data, which resources, # of counters, # of instances, and frequency
- More nodes typically mean more aggregation cost too

Harvest allows you to control which resources and counters to collect, which will help you control the impact, but ultimately you're probably going to need to play with it a bit.

**Reply by MIOKHAN** [reply]
Thanks for your answer.
